cmd/minishift/cmd/start.go

```diff
@@ -127,6 +128,12 @@ func runStart(cmd *cobra.Command, args []string) {
 	setOcProxy()
 	setShellProxy()
 
+	if viper.IsSet("http-proxy") {
```

Comment:
Looking at the proxy logs you post in issue #429, it seems it is very much using HTTPS:

    TCP_TUNNEL/200

So, in this case the proxy serves are relay establishing a tunnel between the subscription-manager and the registration. In this case https flag should be given precedence.